src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/LeftCurlyCheck.java

```diff
@@ -295,6 +297,53 @@ private static DetailAST findLastAnnotation(DetailAST modifiers) {
         return annotation;
     }
 
+    /**
+     * Find brace, considering rules for {@code TokenTypes.LITERAL_ELSE}.
+     * @param ast {@code DetailAST}.
+     * @return {@code DetailAST} brace itself or null if it is not found.
+     */
+    private static DetailAST findBraceForElse(DetailAST ast) {
+        final DetailAST candidate = ast.getFirstChild();
+        DetailAST foundBrace = null;
+
+        if (candidate.getType() == TokenTypes.SLIST) {
+            foundBrace = candidate;
+        }
+        return foundBrace;
+    }
+
+    /**
+     * Find brace, considering rules for
+     * {@code TokenTypes.INTERFACE_DEF}, {@code TokenTypes.CLASS_DEF},
+     * {@code TokenTypes.ANNOTATION_DEF}, {@code TokenTypes.ENUM_DEF},
+     * {@code TokenTypes.ENUM_CONSTANT_DEF}.
+     * @param ast {@code DetailAST}.
+     * @return {@code DetailAST} brace itself or null if it is not found.
+     */
+    private static DetailAST findBraceForClassDefinition(DetailAST ast) {
```

Comment:
please name it findBraceForTypeDefinition , as you process not only classes.
Please be accurate with names.

Reply:
done